### PR TITLE
Add algolia custom search

### LINKED ISSUE #231: Add search functionality in documentation
### Examples from other doc sites

https://chakra-ui.com/

### PR DESCRIPTION
**Pull request by benoitgrelard**
This is a continuation of the work started in #425.
We weren't happy with the default built-in experience of DocSearch, so we built our own experience using lower level libraries from Algolia as well as doing a manual scraping of the docs.

Closes #231

### CODE REVIEW

File: components/PrimitivesDocsPage.tsx [hunk 1]
```diff
@@ -23,6 +23,17 @@ export function PrimitivesDocsPage({ children }: { children: React.ReactNode })
...
+    const mediaQueryList = window.matchMedia('(min-width: 900px)');
+
+    const handleChange = () => {
+      if (mediaQueryList.matches) setIsMobileMenuOpen(false);
+    };
+
+    mediaQueryList.addEventListener('change', handleChange);
+    return () => mediaQueryList.removeEventListener('change', handleChange);
+  }, []);
+
```
Comment (benoitgrelard) [inline components/PrimitivesDocsPage.tsx:36]: We'll need the same in `ColorsDocsPage` too

File: components/PrimitivesDocsSearch.tsx [hunk 2]
```diff
@@ -222,7 +231,6 @@ export function PrimitivesDocsSearch(props: PrimitivesDocsSearchProps) {
           <TextField
             ref={inputRef}
             {...autocomplete.getInputProps({ inputElement: inputRef.current })}
-            type="search"
```
Comment (benoitgrelard) [inline components/PrimitivesDocsSearch.tsx]: Was that intentional? If there's a reason for it, we should remove the unused CSS too then.

File: components/PrimitivesDocsSearch.tsx [hunk 3]
```diff
@@ -283,54 +291,52 @@ export function PrimitivesDocsSearch(props: PrimitivesDocsSearchProps) {
...
-          )}
-        </DismissableLayer>
+        <RemoveScroll as={Slot} allowPinchZoom enabled={!isMobile}>
+          <DismissableLayer
+            asChild
+            disableOutsidePointerEvents={!isMobile}
+            onPointerDownOutside={(event) => {
+              const target = event.target as HTMLElement;
+              const isPointerDownInForm = formRef.current.contains(target);
+              if (!isMobile && !isPointerDownInForm) autocomplete.setIsOpen(false);
```
Comment (benoitgrelard) [inline components/PrimitivesDocsSearch.tsx:301]: 👍

File: package.json [hunk 4]
```diff
@@ -70,7 +71,7 @@
     "prismjs": "^1.23.0",
     "react": "17.0.2",
     "react-dom": "17.0.2",
-    "react-remove-scroll": "^2.4.1",
+    "react-remove-scroll": "^2.5.5",
```
Comment (benoitgrelard) [inline package.json:74]: ♥️

File: components/PrimitivesDocsSearch.tsx (hunk 2 above)
Comment (andy-hook) [reply components/PrimitivesDocsSearch.tsx]: I noticed it's also provided by the prop getter, unless we prefer to redeclare for explicitness?

Comment (benoitgrelard) [reply components/PrimitivesDocsSearch.tsx]: Ah cool, I didn't know that, that's fine if we rely on the prop getter then.

**Review by andy-hook — APPROVED** [review]

**Pull request merged**